storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -2174,17 +2174,19 @@ copy_back(int argc, char **argv)
 bool
 decrypt_decompress_file(const char *filepath, uint thread_n)
 {
-	std::stringstream cmd, message;
-	char buf[FN_LEN];
+	std::stringstream cmd, message,input_file,output_file;
 	bool needs_action = false;
 
-        if (escape_string_for_mysql(&my_charset_utf8mb4_general_ci, buf, 0,
```

Comment:
From our discussion on slack, escape_string_for_mysql() didn't work according to our expectations. Can we fix this instead?

Reply:
Lets take your approach.